src/pcre2_compile_class.c

```diff
@@ -1266,32 +1215,14 @@ while (pptr < end_ptr)
     xclass_props |= XCLASS_HAS_8BIT_CHARS;
 #endif
     continue;               /* End of POSIX handling */
-    }
-
-  /* Other than POSIX classes, the only items we should encounter are
-  \d-type escapes and literal characters (possibly as ranges). */
-
-  if (meta == META_BIGVALUE)
-    {
-    meta = *(pptr++);
-    goto CLASS_LITERAL;
-    }
-
-  /* Any other non-literal must be an escape */
 
-  if (meta >= META_END)
-    {
-    uint32_t escape;
+    /* Other than POSIX classes, the only items we should encounter are
+    \d-type escapes and literal characters (possibly as ranges). */
+    case META_BIGVALUE:
+      meta = *(pptr++);
+      break;
 
-    if (META_CODE(meta) != META_ESCAPE)
-      {
-#ifdef DEBUG_SHOW_PARSED
-      fprintf(stderr, "** Unrecognized parsed pattern item 0x%.8x "
-                      "in character class\n", meta);
-#endif
-      *errorcodeptr = ERR89;  /* Internal error - unrecognized. */
```

Comment:
I think assertions are better for this, so I removed this functionality. If it is desired, it can be added to the call sites of this function.